assets/javascripts/discourse/initializers/chat-setup.js

```diff
@@ -125,6 +129,372 @@ export default {
         });
       });
     });
+
+    withPluginApi("1.3.0", (api) => {
+      api.addSidebarSection(
+        (BaseCustomSidebarSection, BaseCustomSidebarSectionLink) => {
+          const SidebarChatSectionLink = class extends BaseCustomSidebarSectionLink {
+            constructor({ channel, chatService }) {
+              super(...arguments);
+              this.channel = channel;
+              this.chatService = chatService;
+            }
+
+            get name() {
+              return `${this.channel.chatable_type.toLowerCase()}-${
+                this.channel.chatable_id
+              }`;
+            }
+
+            get route() {
+              return "chat.channel";
+            }
+
+            get model() {
+              return {
+                chatChannel: this.channel,
+                channelId: this.channel.id,
+                channelTitle: this.channel.chatable.slug,
+              };
+            }
+
+            get title() {
+              return this.channel.title;
+            }
+
+            get text() {
+              return this.channel.title;
+            }
+
+            get prefixType() {
+              return "icon";
+            }
+
+            get prefixValue() {
+              return "hashtag";
+            }
+
+            get prefixColor() {
+              return this.channel.chatable.color;
+            }
+
+            get prefixBadge() {
+              return this.channel.chatable.read_restricted ? "lock" : "";
+            }
+
+            get suffixType() {
+              return "icon";
+            }
+
+            get suffixValue() {
+              return this.channel.unread_count > 0 ? "circle" : "";
+            }
+
+            get suffixCSSClass() {
+              return this.channel.unread_mentions > 0 ? "urgent" : "unread";
+            }
+
+            get currentWhen() {
+              return (
+                this.chatService.router.currentRouteName?.startsWith(
+                  "chat.channel"
+                ) && this.channel.id === this.chatService.activeChannel?.id
+              );
+            }
```

Comment:
I'm not sure if we need this since the `LinkTo` component should automatically take care of adding the `.active` class to the link.

Reply:
Sidebar link was not active when channel was changed via keyboard shortcut and switch channel modal. This is because switch channel is using params to transitionTo

https://github.com/discourse/discourse-chat/blob/main/assets/javascripts/discourse/services/chat.js#L462

Then active state is calculated here:

https://github.com/discourse/discourse-chat/blob/main/assets/javascripts/discourse/components/chat-channel-row.js#L23

In LinkTo we pass model and active was not correctly detected.

That's why I tried to use `currentWhen` with logic stolen from original sidebar and it solved problem

Reply:
We need to move away from the hacks of the old sidebar. This means that any link present in the sidebar has to be a valid Ember route.